pageserver/src/tenant/storage_layer.rs

```diff
@@ -88,21 +88,34 @@ pub enum ValueReconstructResult {
     Missing,
 }
 
-#[derive(Default)]
+#[derive(Debug)]
 pub struct LayerAccessStats(Mutex<LayerAccessStatsInner>);
 
-#[derive(Clone)]
+#[derive(Debug, Clone)]
 struct LayerAccessStatFullDetails {
     when: SystemTime,
     task_kind: TaskKind,
     access_kind: LayerAccessKind,
 }
 
+#[derive(Debug, Clone)]
 struct LayerAccessStatsInner {
     first_access: Option<LayerAccessStatFullDetails>,
     count_by_access_kind: EnumMap<LayerAccessKind, u64>,
     task_kind_flag: EnumSet<TaskKind>,
     last_accesses: VecDeque<LayerAccessStatFullDetails>,
+    last_residence_changes: VecDeque<LayerResidenceStatus>,
+}
+
+#[derive(Debug, Clone)]
+pub enum LayerResidenceStatus {
+    Resident {
+        timestamp: SystemTime,
+        creating: bool,
```

Comment:
```suggestion
        /// Layer was created during this session.
        created: bool,
```

The `ing` seem like something ongoing, but our file creations should be atomic, layer doesn't really exist before we are done creating and writing a new file.